lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -592,6 +593,9 @@ sub failed_modules_with_needles {
             for my $needle (@{$detail->{needles}}) {
                 push( @needles, [ $needle->{name}, $counter ] );
             }
+            if (!@{$detail->{needles}}) {
+                push(@needles, [ undef, $counter ] );
```

Comment:
Hehe, I noticed. At SLEPOS we used hashref->{'xx'}. But I try to keep your needs at mind. :)